amazon-redshift-plugin/src/main/java/io/cdap/plugin/amazon/redshift/RedshiftDBRecord.java

```diff
@@ -0,0 +1,123 @@
+          field.getSchema().getNonNullable() : field.getSchema();
+        setZonedDateTimeBasedOnOuputSchema(recordBuilder, nonNullableSchema.getLogicalType(),
+                                           field.getName(), zonedDateTime);
+      } else {
+        recordBuilder.set(field.getName(), null);
+      }
+    } else if (sqlType == Types.TIMESTAMP && columnTypeName.equalsIgnoreCase("timestamptz")) {
+      OffsetDateTime timestamp = resultSet.getObject(columnIndex, OffsetDateTime.class);
+      if (timestamp != null) {
+        recordBuilder.setTimestamp(field.getName(), timestamp.atZoneSameInstant(ZoneId.of("UTC")));
+      } else {
+        recordBuilder.set(field.getName(), null);
+      }
+    } else {
+      int columnType = metadata.getColumnType(columnIndex);
+      if (columnType == Types.NUMERIC) {
+        Schema nonNullableSchema = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable() : field.getSchema();
+        int precision = metadata.getPrecision(columnIndex);
+        if (precision == 0 && Schema.Type.STRING.equals(nonNullableSchema.getType())) {
+          // When output schema is set to String for precision less numbers
+          recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
+        } else if (Schema.LogicalType.DECIMAL.equals(nonNullableSchema.getLogicalType())) {
+          BigDecimal orgValue = resultSet.getBigDecimal(columnIndex);
+          if (orgValue != null) {
+            BigDecimal decimalValue = new BigDecimal(orgValue.toPlainString())
+              .setScale(nonNullableSchema.getScale(), RoundingMode.HALF_EVEN);
+            recordBuilder.setDecimal(field.getName(), decimalValue);
+          }
+        }
+        return;
+      }
+      setField(resultSet, recordBuilder, field, columnIndex, sqlType, sqlPrecision, sqlScale);
+    }
+  }
+
+  private void setZonedDateTimeBasedOnOuputSchema(StructuredRecord.Builder recordBuilder,
+                                                  Schema.LogicalType logicalType,
+                                                  String fieldName,
+                                                  ZonedDateTime zonedDateTime) {
+    if (Schema.LogicalType.DATETIME.equals(logicalType)) {
+      recordBuilder.setDateTime(fieldName, zonedDateTime.toLocalDateTime());
+    } else if (Schema.LogicalType.TIMESTAMP_MICROS.equals(logicalType)) {
+      recordBuilder.setTimestamp(fieldName, zonedDateTime);
+    }
+  }
+
+  private static boolean isUseSchema(ResultSetMetaData metadata, int columnIndex) throws SQLException {
+    String columnTypeName = metadata.getColumnTypeName(columnIndex);
+    // If the column Type Name is present in the String mapped Redshift types then return true.
+    if (RedshiftSchemaReader.STRING_MAPPED_REDSHIFT_TYPES_NAMES.contains(columnTypeName)
+      || RedshiftSchemaReader.STRING_MAPPED_REDSHIFT_TYPES.contains(metadata.getColumnType(columnIndex))) {
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
Returning a boolean after evaluation of an boolean expression can usually be reduced to returning the boolean expression.

---
Suggestion
```suggestion
    // If the column Type Name is present in the String mapped Redshift types then return true.
return PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES_NAMES.contains(columnTypeName)
            || PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES.contains(metadata.getColumnType(columnIndex));

```